### PR TITLE
update documentation for maintainers

### PR DESCRIPTION
**Pull request by kejxu**
this is just a documentation update, no functionality change

also expecting to merge this commit into https://github.com/ms-iot/vscode-ros/tree/release/1906

### CODE REVIEW

**Comment by seanyen** [comment]
Is there any noticeable changes to review (things are really deleted or added) or it simply move around the existing document into more proper sections?

**Comment by kejxu** [comment]
> Is there any noticeable changes to review (things are really deleted or added) or it simply move around the existing document into more proper sections?

Thanks for pointing this out! Aside from the diffs created by moving documentation around, there are some new documentation added:
- https://github.com/ms-iot/vscode-ros/pull/69/files#diff-979ef53b520be22ab13fcd2253dcc020R30
- https://github.com/ms-iot/vscode-ros/pull/69/files#diff-979ef53b520be22ab13fcd2253dcc020R48
- https://github.com/ms-iot/vscode-ros/pull/69/files#diff-979ef53b520be22ab13fcd2253dcc020R108

File: MAINTAINERS_GUIDELINES.md [hunk 1]
```diff
@@ -59,37 +80,65 @@ Please review the following before publishing a new release:
...
+The release process can be automated by triggering a release pipeline from the [vscode-ros.release pipeline][vscode-ros.release]; however, we will be simplifying the process to only release manually for now.
+
+To authorize a release manually, schedule a release build with the proper release branch for the [vscode-ros.release pipeline][vscode-ros.release].
 
 ![schedule a release build][schedule_manual_release_build]
 
-#### Triggering an automatic release (with a new tag)
+After the extension is released, the release branch should be:
+
+1. tagged with version number
```
Comment (seanyen) [inline MAINTAINERS_GUIDELINES.md]: In this case, the git graph will be like:
image

Is it the expected result per the steps here? I cannot quickly think of any big issues right now but just check that's the desired tree after this operations.

Comment (kejxu) [reply MAINTAINERS_GUIDELINES.md]: I think this would be fine.

**Pull request merged**